scripts/liquidation_bot/index.ts

```diff
@@ -11,21 +11,24 @@ import {
   Asset
 } from './liquidateUnderwaterBorrowers';
 import { FlashbotsBundleProvider } from '@flashbots/ethers-provider-bundle';
-import { providers, Wallet } from 'ethers';
+import { Signer, Wallet } from 'ethers';
 import googleCloudLog, { LogSeverity } from './googleCloudLog';
 
 const loopDelay = 5000;
 const loopsUntilUpdateAssets = 1000;
 let assets: Asset[] = [];
 
 async function main() {
-  let { DEPLOYMENT: deployment, LIQUIDATOR_ADDRESS: liquidatorAddress, USE_FLASHBOTS: useFlashbots } = process.env;
+  let { DEPLOYMENT: deployment, LIQUIDATOR_ADDRESS: liquidatorAddress, USE_FLASHBOTS: useFlashbots, ETH_PK: ethPk } = process.env;
   if (!liquidatorAddress) {
     throw new Error('missing required env variable: LIQUIDATOR_ADDRESS');
   }
   if (!deployment) {
     throw new Error('missing required env variable: DEPLOYMENT');
   }
+  if (!ethPk) {
```

Comment:
`ethPk` is only required if `useFlashbots` is true, right?

Reply:
> You mentioned that, when using Flashbots, the transaction will fail silently (it just won't get picked up by Flashbots and after some time we'll assume that it has failed).
> 
> What about the non-Flashbots case? Are transaction failures immediate?

Actually, I figured out a way to block on the Flashbots txn and get a response on block inclusion or not. So we can now add alerting if a txn does not get included X amount of times (but probably better to just alert if there's purchaseable collateral available for X consecutive blocks).

Transaction failures are never silent. If a txn would fail, Hardhat would always convey that.

Reply:
Yeah that's correct. I can change this check.

Reply:
it's up to you; the Pulumi configs all contain an `ETH_PK` right now, so it's not a huge deal

Reply:
I updated the check because it's strictly only needed when `useFlashbots` is true.